src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java

```diff
@@ -260,6 +260,50 @@ boolean onlyMinorChanges(AbstractGroup oldGroup, AbstractGroup newGroup) {
         return true;
     }
 
+    /**
+     * Opens "Rename Group Dialog" and change name of group
+     */
+    public void renameGroup(GroupNodeViewModel oldGroup) {
+        currentDatabase.ifPresent(database -> {
+            AbstractGroup oldGroupDef = oldGroup.getGroupNode().getGroup();
+            String oldGroupName = oldGroupDef.getName();
+
+            Optional<AbstractGroup> newGroup = dialogService.showCustomDialogAndWait(
+                    new RenameGroupView(database,
+                            oldGroup.getGroupNode().getGroup())
+            );
+
+            newGroup.ifPresent(group -> {
+
+                String newGroupName = group.getName();
+
+                if (newGroupName.trim().isEmpty()) {
+                    return;
+                }
+
+                if (newGroupName.contains(",")) {
+                    return;
+                }
+
+                if (oldGroupName.equals(newGroupName)) {
+                    return;
+                }
+
+                int groupsWithSameName = 0;
+                Optional<GroupTreeNode> databaseRootGroup = currentDatabase.get().getMetaData().getGroups();
+                if (databaseRootGroup.isPresent()) {
+                    groupsWithSameName = databaseRootGroup.get().findChildrenSatisfying(g -> g.getName().equals(newGroupName)).size();
```

Comment:
There needs to be comment why this is necessary. At first guess, I would have thought that there must not be any other group with the same name. Why is one other group with the same name OK?